File: Js/S2_FOR_06.js
function sumNested(matrix) {
    let total = 0;
    for (let row of matrix) {
        for (let num of row) {
            total += num;
        }
    }
    return total;
}

console.log(sumNested([[1, 2], [3, 4], [5]]));  // 15
console.log(sumNested([[1, 2, 3], [4, 5], [6]]));  // 21

File: Python/Stage 2/S2_FOR_06.py
def sum_nested(matrix):
    total = 0
    for row in matrix:
        if not isinstance(row, list):
            return None
        for value in row:
            total += value
    return total


print(sum_nested([[1, 2], [3, 4]]))
print(sum_nested([[1], "x", [2]]))
